lib/wasmex/instance.ex

```diff
@@ -61,17 +63,22 @@ defmodule Wasmex.Instance do
       ...>       "imported_void" => {:fn, [], [], fn _context -> nil end}
       ...>     }
       ...> }
-      iex> {:ok, %Wasmex.Instance{}} = Wasmex.Instance.new(store, module, imports)
+      ...> links = []
+      iex> {:ok, %Wasmex.Instance{}} = Wasmex.Instance.new(store, module, imports, links)
   """
-  @spec new(Wasmex.StoreOrCaller.t(), Wasmex.Module.t(), %{
-          optional(binary()) => (... -> any())
-        }) ::
+  @spec new(
+          Wasmex.StoreOrCaller.t(),
+          Wasmex.Module.t(),
+          %{optional(binary()) => (... -> any())},
+          [%{optional(binary()) => Wasmex.Module.t()}] | []
```

Comment:
hmm, it's not a Wasmex.Module we get here, right? But a modules `resource` if I'm not mistaken from just reading the source above. What do you think of changing `Wasmex.build_compiled_links` so that it doesn't return a list of module-resources but actual module structs?

Reply:
thinking about it: 

* I like having actual module structs in the elixir API not exposing the inner `resource` (which holds the rust struct) if possible
* all kinds of module functions we expose should theoretically work then without trouble
* downside is that we have to iterate over each link-module here in the `new` function, mapping them to their internal `resources`.